missing entries in libtbb.so for armv7 tbb package ?

While packaging oce (PR #4270), It fails on linking with libtbb.so (package tbb) while cross-building for armv7hf (at least), whether glibc or musl.
Same packaging build  but is fine for x86_64 and x86, whether glibc or musl.

It seems that the `libtbb.so` does not contains the same entry, hence the failed linking below:
- on x86_64 glibc:  
  
  ```
  $ file /usr/lib/libtbb.so.2
  /usr/lib/libtbb.so.2: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=1f16233f751b13a6d5b4843ed8ea69042ead63d6, stripped
  
  $ objdump -T /usr/lib/libtbb.so.2 | grep allocate_child_proxy
  000000000001c6f0 g    DF .text  0000000000000053  Base        _ZNK3tbb8internal20allocate_child_proxy4freeERNS_4taskE
  000000000001c6a0 g    DF .text  000000000000004a  Base        _ZNK3tbb8internal20allocate_child_proxy8allocateEm
  
  ```
- the same lib on armv7 :  
  
  ```
  $file /build/packages/masterdir.armv7hf/usr/armv7l-linux-gnueabihf/usr/lib/libtbb.so.2
  /build/packages/masterdir.armv7hf/usr/armv7l-linux-gnueabihf/usr/lib/libtbb.so.2: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, BuildID[sha1]=622323e98142d55a8d679209d02139e396cbe5a0, stripped
  
  $ objdump -T /build/packages/masterdir.armv7hf/usr/armv7l-linux-gnueabihf/usr/lib/libtbb.so.2  | grep allocate_child_proxy
  0001afac g    DF .text  0000006c  Base        _ZNK3tbb8internal20allocate_child_proxy4freeERNS_4taskE
  ```

The Initial build error message:  
https://travis-ci.org/voidlinux/void-packages/jobs/136426588

Same build error (other build) , with no parallel jobs and cmake option `-DCMAKE_VERBOSE_MAKEFILE=ON`:  

```
[ 99%] Linking CXX executable DRAWEXE
cd /builddir/oce-OCE-0.17.1/build/adm/cmake/DRAWEXE && /usr/bin/cmake -E cmake_link_script CMakeFiles/DRAWEXE.dir/link.txt --verbose=1
/builddir/.xbps-oce/wrappers/armv7l-linux-gnueabihf-g++   -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1 -fstack-protector-strong -D_FORTIFY_SOURCE=2  -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include  -O3 -DNDEBUG   -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed     -L/usr/armv7l-linux-gnueabihf/usr/lib CMakeFiles/DRAWEXE.dir/__/__/__/src/DRAWEXE/DRAWEXE.cxx.o  -o DRAWEXE -rdynamic ../../../Unix/-Release-32/libTKDraw.so.10.0.0 ../../../Unix/-Release-32/libTKMesh.so.10.0.0 ../../../Unix/-Release-32/libTKHLR.so.10.0.0 ../../../Unix/-Release-32/libTKService.so.10.0.0 ../../../Unix/-Release-32/libTKShHealing.so.10.0.0 ../../../Unix/-Release-32/libTKTopAlgo.so.10.0.0 ../../../Unix/-Release-32/libTKGeomAlgo.so.10.0.0 ../../../Unix/-Release-32/libTKBRep.so.10.0.0 ../../../Unix/-Release-32/libTKGeomBase.so.10.0.0 ../../../Unix/-Release-32/libTKG3d.so.10.0.0 ../../../Unix/-Release-32/libTKG2d.so.10.0.0 ../../../Unix/-Release-32/libTKMath.so.10.0.0 -ltbb ../../../Unix/-Release-32/libTKernel.so.10.0.0 -lpthread -ldl -lm -ltbbmalloc -lSM -lICE -lXext -lGLU -lGL -lGLU -lGL -lfreetype -ltcl8.6 -ltk8.6 -lX11 -Wl,-rpath,/builddir/oce-OCE-0.17.1/build/Unix/-Release-32: 
../../../Unix/-Release-32/libTKMesh.so.10.0.0: undefined reference to `tbb::internal::allocate_child_proxy::allocate(unsigned int) const'
../../../Unix/-Release-32/libTKMesh.so.10.0.0: undefined reference to `tbb::internal::allocate_continuation_proxy::allocate(unsigned int) const'
../../../Unix/-Release-32/libTKMath.so.10.0.0: undefined reference to `tbb::internal::allocate_root_proxy::allocate(unsigned int)'
../../../Unix/-Release-32/libTKMesh.so.10.0.0: undefined reference to `tbb::internal::allocate_root_with_context_proxy::allocate(unsigned int) const'
collect2: error: ld returned 1 exit status
```

--- a/srcpkgs/tbb/patches/hostname-usage.patch
+++ b/srcpkgs/tbb/patches/hostname-usage.patch
@@ -1,0 +1,11 @@
+--- build/version_info_linux.sh.ORIG
++++ build/version_info_linux.sh
+@@ -20,7 +20,7 @@
+ 
+ # Script used to generate version info string
+ echo "#define __TBB_VERSION_STRINGS(N) \\"
+-echo '#N": BUILD_HOST'"\t\t"`hostname -s`" ("`uname -m`")"'" ENDL \'
++echo '#N": BUILD_HOST'"\t\t"`hostname`" ("`uname -m`")"'" ENDL \'
+ # find OS name in *-release and issue* files by filtering blank lines and lsb-release content out
+ echo '#N": BUILD_OS'"\t\t"`lsb_release -sd 2>/dev/null | grep -ih '[a-z] ' - /etc/*release /etc/issue 2>/dev/null | head -1 | sed -e 's/["\\\\]//g'`'" ENDL \'
+ echo '#N": BUILD_KERNEL'"\t"`uname -srv`'" ENDL \'

--- a/srcpkgs/tbb/template
+++ b/srcpkgs/tbb/template
@@ -1,27 +1,34 @@
 # Template file for 'tbb'
 pkgname=tbb
-version=2017.20161128
+_ver=2018
+version=${_ver/_U/.}
 revision=1
-_lv="${version%.*}"
-_rv="${version##*.}"
-wrksrc="$pkgname${_lv}_${_rv}oss"
+wrksrc="${pkgname}-${_ver}"
 build_style=gnu-makefile
 short_desc="Intel Threading Building Blocks"
 maintainer="Andrea Brancaleoni <abc@pompel.me>"
 license="Apache-2.0"
 homepage="http://www.threadingbuildingblocks.org"
-distfiles="https://www.threadingbuildingblocks.org/sites/default/files/software_releases/source/tbb${_lv}_${_rv}oss_src.tgz"
-checksum=c009166233c8ea0e34530a1c5f870b79314316d19e6876b37a7e7c854080a540
+distfiles="https://github.com/01org/tbb/archive/${_ver}.tar.gz"
+checksum=94f643f1edfaccb57d64b503c7c96f00dec64e8635c054bbaa33855d72c5822d
 
 only_for_archs="i686-musl x86_64-musl armv7l-musl aarch64-musl i686 x86_64 armv7l aarch64"
+
+case "$XBPS_TARGET_MACHINE" in
+	armv7l*)
+		make_build_args="arch=armv7"
+		;;
+	aarch64*)
+		make_build_args="arch=arm64"
+		;;
+esac
 
 post_extract() {
 	sed -i 's|CPLUS = g..|CPLUS = $(CXX)|g' build/linux.gcc.inc
 	sed -i 's|CONLY = gcc|CONLY = $(CC)|g' build/linux.gcc.inc
-	if [ "$CROSS_BUILD" ]; then
-		sed -i '/-m64/d' build/linux.gcc.inc
-		sed -i '/-mrtm/d' build/linux.gcc.inc
-	fi
+
+	# alternative might be:
+	# https://git.alpinelinux.org/cgit/aports/tree/testing/libtbb/glibc-struct-mallinfo.patch
 	case "$XBPS_TARGET_MACHINE" in
 		*-musl)
 			sed -e "s@#define MALLOC_UNIXLIKE_OVERLOAD_ENABLED __linux__@@" \

--- a/srcpkgs/tbb/update
+++ b/srcpkgs/tbb/update
@@ -1,3 +1,2 @@
-site='https://www.threadingbuildingblocks.org/download'
-pattern='tbb\d+_\K\d+'
-version="$_rv"
+pattern="/01org/tbb/archive/\K[0-9_U]+(?=\.tar\.gz)"
+version="${_ver}"

